### PR TITLE
add dropout option act like tf

### PR DESCRIPTION
**Pull request by phlrain**
1. add dropout option act like tf
2. add float64 supprt for transpose op and softmax_cudnn_op
3. fix bug in global norm clip

### CODE REVIEW

**Comment by phlrain** [comment]
给dropout 增加一个新的属性 div_prob_in_train，默认值为false，如果值为false的时候，和之前表现一样，如果只为true的时候，会在训练的时候 div / 1-p ；解码的时候不做任何处理

**Comment by panyx0718** [comment]
It seems you are doing 3 logically different things in 1 PR. Can you split it into 3?

File: paddle/fluid/operators/dropout_op.cu [hunk 1]
```diff
@@ -47,7 +47,11 @@ __global__ void RandomGenerator(const size_t n, const int seed,
     if (dist(rng) < dropout_prob) {
       mask = static_cast<T>(0);
     } else {
-      mask = static_cast<T>(1);
+      if (div_prob_in_train) {
+        mask = static_cast<T>(1.0f / (1.0f - dropout_prob));
```
Comment (chengduoZH) [inline paddle/fluid/operators/dropout_op.cu:52]: `div` is time consuming, so, you had better precompute `static_cast<T>(1.0f / (1.0f - dropout_prob))` outside of `for`.

File: paddle/fluid/operators/dropout_op.cu [hunk 2]
```diff
@@ -67,6 +71,8 @@ class GPUDropoutKernel : public framework::OpKernel<T> {
     y->mutable_data<T>(context.GetPlace());
     float dropout_prob = context.Attr<float>("dropout_prob");
 
+    auto div_prob_in_train = context.Attr<bool>("div_prob_in_train");
+    VLOG(10) << "div prob in tain" << div_prob_in_train;
```
Comment (chengduoZH) [inline paddle/fluid/operators/dropout_op.cu]: seems that line 75 is unnecessary.

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -974,7 +974,12 @@ def cos_sim(X, Y):
     return out
 
 
-def dropout(x, dropout_prob, is_test=False, seed=None, name=None):
+def dropout(x,
+            dropout_prob,
+            is_test=False,
+            seed=None,
+            name=None,
+            div_prob_in_train=False):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: Betther to tell users the calculation formulas when set div_prob_in_train True  and False. And in the doc, tell users when set True, the behavior is the same as TensorFlow ?

**Comment by panyx0718** [comment]
We can add an iCafe project to revisit this in 1.2. tf's implementation can skip dropout at inference time, which can be a performance gain
